### PR TITLE
Creating cluster assignments and calculating for dataset for IL2

### PR DESCRIPTION
**Pull request by andrewram4287**

### CODE REVIEW

File: gmm/imports.py [hunk 1]
```diff
@@ -33,19 +34,21 @@ def smallDF(numCells: int):
...
+    celltypeXA = flowXA["Cell Type"]
+    flowXA = flowXA.drop_vars(["Cell Type"])
+    flowXA = flowXA[transCols].to_array(dim="Marker")
     # Final Xarray has dimensions [Marker, Cell Number, Time, Dose, Ligand]
+  
 
-    assert np.all(np.isfinite(flowDF.to_numpy()))
-    return flowDF, (experimentcells, cell_type)
+    assert np.all(np.isfinite(flowXA.to_numpy()))
+    return flowXA, celltypeXA, flowDF
```
Comment (aarmey) [inline gmm/imports.py:54]: This almost certainly breaks other code. Have you checked that this builds and passes tests?

File: gmm/imports.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 """ Methods for data import and normalization. """
 
+from types import CellType
```
Comment (aarmey) [inline gmm/imports.py:3]: Doesn't look like you use this import?

File: gmm/figures/figure4.py [hunk 3]
```diff
@@ -68,3 +68,34 @@ def makeFigure():
...
+    resps = cell_assignment(drugXA.to_numpy(), fac)
+    resps = resps / np.reshape(np.sum(resps, axis=1), (-1, 1, drugXA.shape[2], drugXA.shape[3], drugXA.shape[4]))
+    types = typeXA.to_numpy()
+
+    totalshapes = [(drugXA.shape[1]*resps.shape[3]*resps.shape[4]), (drugXA.shape[1]*resps.shape[2]*resps.shape[4]), (drugXA.shape[1]*resps.shape[2]*resps.shape[3])]
+    conditions = {"Time": np.repeat(flowDF["Time"].unique(), totalshapes[0]), "Dose": np.repeat(flowDF["Dose"].unique(),  totalshapes[1]),
+                  "Ligand": np.repeat(flowDF["Ligand"].unique(), totalshapes[2])}
+    typesDF = pd.DataFrame({"Cell Type": np.ravel(types), **conditions})
+    
+    for i in range(0, resps.shape[1]):
```
Comment (aarmey) [inline gmm/figures/figure4.py:85]: Can you comment on what this is doing? There should be a way to go about it without all the loops...

**Review by aarmey — CHANGES_REQUESTED** [review]

File: gmm/figures/figure4.py (hunk 3 above)
Comment (andrewram4287) [reply gmm/figures/figure4.py:85]: Sounds good working on it!

Comment (andrewram4287) [reply gmm/figures/figure4.py]: @aarmey How do i ensure that a cluster is assigned to a specific cell type based on the responsibilities? Is the only way to do this via adj_rand_score?

Comment (aarmey) [reply gmm/figures/figure4.py]: I think you should first just make a [confusion matrix](https://en.wikipedia.org/wiki/Confusion_matrix) between manually gating and the clusters.

Comment (andrewram4287) [reply gmm/figures/figure4.py]: @aarmey I am still having trouble how we can compare cluster assignments from the responsibilities to the actual true cell type label. I understand it logically but I dont understand how to implement the process

Comment (aarmey) [reply gmm/figures/figure4.py]: @borcuttjahns I think you had this figured out?

Comment (andrewram4287) [reply gmm/figures/figure4.py]: I am implemented Brians implementation before don't know a better way of going about the process without all the for loops

Comment (aarmey) [reply gmm/figures/figure4.py]: Oh, apologies, I could have been clearer. I meant that I would look into something without the for loops.

File: gmm/figures/figure4.py [hunk 4]
```diff
@@ -68,3 +67,26 @@ def makeFigure():
         ax[i + 6].set(title="Covariance: Rank - " + str(i + 1))
 
     return f
+    
+def cluster_type(drugXA, fac, typeXA, ax):
```
Comment (andrewram4287) [inline gmm/figures/figure4.py:70]: @aarmey Is this method better than the previous for comparing the predicted and actual cell assignments

Comment (aarmey) [reply gmm/figures/figure4.py:70]: Yes!! You can use the `axis` argument of `argmax()` to get rid of the loops.

**Comment by andrewram4287** [comment]
@aarmey I also added a bit more of an efficient soft clustering methods as well, other than that the PR should be ready to merge!

**Review by aarmey — APPROVED** [review]
Check that this builds. Otherwise, this looks ready to merge to me!

**Pull request merged**